pennylane/io.py

```diff
@@ -410,113 +407,131 @@ def from_qiskit_op(qiskit_op, params=None, wires=None):
+            measurement or list of PennyLane measurements that overrides the terminal measurements
+            that may be present in the input circuit. Defaults to ``None``, such that existing measurements
+            in the input circuit are kept. Mid-circuit measurements are kept either way.
 
     Returns:
-        function: the PennyLane template created based on the QASM string
+        function: the PennyLane quantum function created based on the QASM string. This function itself returns the mid-circuit measurements plus the terminal measurements by default (``measurements=None``), and returns **only** the measurements from the ``measurements`` argument otherwise.
 
-    By default, ``from_qasm`` will remove any measurements that are present in the QASM code.
-    If the QASM code contains measurements, set ``measurements=None`` to keep them in the
-    output of ``from_qasm``.
+    **Example:**
 
-    .. warning::
+    .. code-block:: python
 
-        The current default behaviour of removing measurements in the QASM code is deprecated
-        and will be changed in a future release. Starting in version ``0.38``, ``from_qasm``
-        will keep the measurements from the QASM code by default. To remove all measurements,
-        set ``measurements=[]`` which overrides the existing measurements with an empty list.
+        qasm_code = 'OPENQASM 2.0;'\\
+                    'include "qelib1.inc";'\\
+                    'qreg q[2];'\\
+                    'creg c[2];'\\
+                    'h q[0];'\\
+                    'measure q[0] -> c[0];'\\
+                    'rz(0.24) q[0];'\\
+                    'cx q[0], q[1];'\\
+                    'measure q -> c;'
 
-    **Example:**
+        loaded_circuit = qml.from_qasm(qasm_code)
+
+    >>> print(qml.draw(loaded_circuit)())
+    0: ──H──┤↗├──RZ(0.24)─╭●──┤↗├─┤
+    1: ───────────────────╰X──┤↗├─┤
+
+    Calling the quantum function returns a tuple containing the mid-circuit measurements and the terminal measurements.
+
+    >>> my_circuit()
+    (MeasurementValue(wires=[0]),
+    MeasurementValue(wires=[0]),
+    MeasurementValue(wires=[1]))
+
+    To remove all terminal measurements, set ``measurements=[]`` which removes the existing terminal measurements, while keeping the mid circuit measurements.
 
     .. code-block:: python
 
-        >>> hadamard_qasm = 'OPENQASM 2.0;' \\
-        ...                 'include "qelib1.inc";' \\
-        ...                 'qreg q[1];' \\
-        ...                 'creg c[1];' \\
-        ...                 'h q[0];'
-        >>> my_circuit = qml.from_qasm(hadamard_qasm)
-        >>> my_circuit()
-        []
+        loaded_circuit = qml.from_qasm(qasm_code, measurements=[])
+
+    >>> print(qml.draw(loaded_circuit)())
+    0: ──H──┤↗├──RZ(0.24)─╭●─┤
+    1: ───────────────────╰X─┤
+
+    Note that calling the quantum function returns the same empty list we passed.
+
+    >>> loaded_circuit()
+    []
```

Comment:
I'd be tempted to make this a separate expandable section. E.g., "Removing terminal measurements". Otherwise, it feels like we're maybe overexplaining this part which is probably an edge case for most users.

Reply:
I was going to say the same thing - since the `[]` input really just exists to allow people to access legacy behaviour, it feels like there's no reason to make the general explanation more confusing by mentioning it. Keeping it separate also limits how much we encourage people to do it, in case we do want to eventually deprecate and remove the legacy behaviour entirely.